writer.go

```diff
@@ -903,6 +916,9 @@ func (w *writer) writeFileFooter() error {
 }
 
 func (w *writer) writeRowGroup(rowGroupSchema *Schema, rowGroupSortingColumns []SortingColumn) (int64, error) {
+	if len(w.columns) == 0 {
+		return 0, nil
+	}
 	numRows := w.columns[0].totalRowCount()
```

Comment:
This was the actual source of the panic. With the three lines added above, we can write data to the file and close it without panicking... But also without doing anything since the file will always be empty if the schema has no columns...